webdriver/tests/bidi/browsing_context/set_viewport/invalid.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+import webdriver.bidi.error as error
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+@pytest.mark.asyncio
+async def test_params_context_invalid_context(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.set_viewport(context=value, viewport={
+            "width": 100,
+            "height": 200,
+        })
+
+
+@pytest.mark.parametrize("viewport", [{"width": -1, "height": -1}, {"width": None, "height": 300}, {"width": 1200, "height": "String"}])
+@pytest.mark.asyncio
+async def test_params_context_invalid_viewport(bidi_session, new_tab, viewport):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.set_viewport(context=new_tab["context"], viewport=viewport)
+
+
+@pytest.mark.asyncio
+async def test_invalid_frame(bidi_session):
+    with pytest.raises(error.NoSuchFrameException):
+        await bidi_session.browsing_context.set_viewport(context="_invalid_")
+
+
+@pytest.mark.asyncio
+async def test_closed_frame(bidi_session, top_context, inline, add_and_remove_iframe):
```

Comment:
This test was added only once so far for screenshots. No other API has it yet. I would say that we add it later on for all the other APIs in on go? 

Instead I would propose a test that creates a frame but doesn't remove it. Calling `setViewport` for the frame has to fail because we only allow top-level browsing contexts to be modified.

Reply:
I will remove the test and do a new test separately. The fixture for adding/removing frames needs refactoring as it is combined in one for some reason.